app/controllers/consent_forms/edit_controller.rb

```diff
@@ -73,6 +77,15 @@ def set_consent_form
     @consent_form = ConsentForm.find(params.fetch(:consent_form_id))
   end
 
+  def set_steps
+    # Translated steps are cached after running setup_wizard_translated.
+    # To allow us to run this method multiple times during a single action
+    # lifecycle, we need to clear the cache.
+    @wizard_translations = nil
```

Comment:
The reason we need this is because the library is setting `@wizard_translation ||=`: https://github.com/zombocom/wicked/blob/7037c81780cac6eae0ed2d51560eafa91651e5e4/lib/wicked/wizard/translated.rb#L28

During an `update` action, the `before_action`s get triggered, setting the form steps to an array of `form_steps` that don't contain `reason`.

After `@consent_form.assign_attributes(update_params)` in the update action, the `form_steps` are updated on the instance, but the internal state of the Wizard is still operating on the old `form_steps`. Setting `self.steps` isn't enough, we also need to trigger `setup_wizard_translated` again and in addition force the updating of the `@wizard_translations` array by setting it to `nil` before.

There shouldn't be much downside in re-running these operations twice for update actions, other than the fact that it's not super clean.